src/tests/unit/tests/DetailsView/components/requirement-view.test.tsx

```diff
@@ -40,19 +64,100 @@ describe('RequirementViewTest', () => {
                 ),
             )
             .returns(() => selectedRequirementStub);
+        assessmentInstanceTableHandlerStub = {
+            changeRequirementStatus: null,
+        } as AssessmentInstanceTableHandler;
+        manualRequirementResultMapStub = {
+            'some manual test step result id': null,
+        };
+        featureFlagStoreDataStub = {
+            'some feature flag': true,
+        };
+        pathSnippetStoreDataStub = {
+            path: null,
+        } as PathSnippetStoreData;
+        updateHandlerMock = Mock.ofType(AssessmentViewUpdateHandler);
 
-        const props: RequirementViewProps = {
-            deps: {} as RequirementViewDeps,
+        props = {
+            deps: { assessmentViewUpdateHandler: updateHandlerMock.object } as RequirementViewDeps,
             requirement: requirementStub,
             assessmentsProvider: assessmentsProviderMock.object,
             assessmentNavState: assessmentNavState,
             instancesMap: {} as DictionaryStringTo<GeneratedAssessmentInstance>,
-            isStepEnabled: true,
-            isStepScanned: true,
-        };
+            isRequirementEnabled: true,
+            isRequirementScanned: true,
+            manualRequirementResultMap: manualRequirementResultMapStub,
+            assessmentInstanceTableHandler: assessmentInstanceTableHandlerStub,
+            featureFlagStoreData: featureFlagStoreDataStub,
+            pathSnippetStoreData: pathSnippetStoreDataStub,
+            prevTarget: { id: 4 },
+            currentTarget: { id: 5 },
+            assessmentData: {},
+        } as RequirementViewProps;
+    });
 
+    it('renders with content from props', () => {
         const rendered = shallow(<RequirementView {...props} />);
 
         expect(rendered.getElement()).toMatchSnapshot();
     });
+
+    test('componentDidUpdate', () => {
+        const newProps = cloneDeep(props);
+        newProps.deps.assessmentViewUpdateHandler = updateHandlerMock.object;
```

Comment:
I'm curious why we have to set the `assessmentViewUpdateHandler` here after `cloneDeep`?

Reply:
Clone deep won't/can't clone the functional properties on an object. That is to say: the mock would need to be either re-created or in this case, re-assigned.